src/components/dashboard/main-content/course-enrollments/course-cards/unenroll/UnenrollModal.jsx

```diff
@@ -33,6 +40,33 @@ const UnenrollModal = ({
     onClose();
   };
 
+  const updateQueryForUnenrollment = () => {
+    const enrollmentForCourseFilter = (enrollment) => enrollment.courseRunId !== courseRunId;
+    const isBFFEnabled = isBFFEnabledForEnterpriseCustomer(enterpriseCustomer.uuid);
+    // Determine which BFF queries need to be updated after unenrolling.
+    if (isBFFEnabled) {
+      const dashboardBFFQueryKey = queryEnterpriseLearnerDashboardBFF(
+        { enterpriseSlug: params.enterpriseSlug },
+      ).queryKey;
+      const bffQueryKeysToUpdate = [dashboardBFFQueryKey];
+      // Update the enterpriseCourseEnrollments data in the cache for each BFF query.
+      bffQueryKeysToUpdate.forEach((queryKey) => {
+        const existingBFFData = queryClient.getQueryData(queryKey);
+        const updatedBFFData = {
+          ...existingBFFData,
+          enterpriseCourseEnrollments: existingBFFData.enterpriseCourseEnrollments.filter(enrollmentForCourseFilter),
+        };
```

Comment:
Per our discussion yesterday, it still might be worth being defensive here to only attempt `setQueryData` when `existingBFFData` exists to prevent potential JS errors, e.g.

```jsx
const existingBFFData = queryClient.getQueryData(queryKey);
if (!existingBFFData) {
  logInfo(`Skipping optimistic cache update of {queryKey} as no cached query data exists yet.`);
  return;
}
const updatedBFFData = {
  ...existingBFFData,
  enterpriseCourseEnrollments: existingBFFData.enterpriseCourseEnrollments.filter(enrollmentForCourseFilter),
};
```